File: .sdk-repos/orchestration-cluster-api-go/examples/decision.go
// Decision operations: evaluate, and read decision definitions, instances, and
// requirements (DRDs).
package examples

import (
	"context"
	"fmt"

	camunda "github.com/camunda/orchestration-cluster-api-go"
	openapi "github.com/camunda/orchestration-cluster-api-go/client"
)

func evaluateDecisionExample(ctx context.Context, client *camunda.CamundaClient) error {
	// region EvaluateDecision
	// DecisionEvaluationInstruction is a union; evaluate by decision id here.
	byID := openapi.NewDecisionEvaluationById("dish-decision")
	byID.SetVariables(map[string]any{"season": "Winter", "guestCount": 4})

	result, err := client.EvaluateDecision(ctx,
		openapi.DecisionEvaluationByIdAsDecisionEvaluationInstruction(byID))
	if err != nil {
		return err
	}
	fmt.Printf("%v\n", result)
	// endregion EvaluateDecision
	return nil
}

func searchDecisionDefinitionsExample(ctx context.Context, client *camunda.CamundaClient) error {
	// region SearchDecisionDefinitions
	result, err := client.SearchDecisionDefinitions(ctx, *openapi.NewDecisionDefinitionSearchQuery())
	if err != nil {
		return err
	}
	for _, d := range result.GetItems() {
		fmt.Printf("%v\n", d)
	}
	// endregion SearchDecisionDefinitions
	return nil
}

func getDecisionDefinitionExample(ctx context.Context, client *camunda.CamundaClient) error {
	// region GetDecisionDefinition
	def, err := client.GetDecisionDefinition(ctx, openapi.MustDecisionDefinitionKey("2251799813685310"))
	if err != nil {
		return err
	}
	fmt.Printf("%v\n", def)
	// endregion GetDecisionDefinition
	return nil
}

func getDecisionDefinitionXMLExample(ctx context.Context, client *camunda.CamundaClient) error {
	// region GetDecisionDefinitionXML
	xml, err := client.GetDecisionDefinitionXML(ctx, openapi.MustDecisionDefinitionKey("2251799813685310"))
	if err != nil {
		return err
	}
	fmt.Println(xml)
	// endregion GetDecisionDefinitionXML
	return nil
}

func searchDecisionInstancesExample(ctx context.Context, client *camunda.CamundaClient) error {
	// region SearchDecisionInstances
	result, err := client.SearchDecisionInstances(ctx, *openapi.NewDecisionInstanceSearchQuery())
	if err != nil {
		return err
	}
	for _, d := range result.GetItems() {
		fmt.Printf("%v\n", d)
	}
	// endregion SearchDecisionInstances
	return nil
}

func getDecisionInstanceExample(ctx context.Context, client *camunda.CamundaClient) error {
	// region GetDecisionInstance
	instance, err := client.GetDecisionInstance(ctx, "2251799813685310-1")
	if err != nil {
		return err
	}
	fmt.Printf("%v\n", instance)
	// endregion GetDecisionInstance
	return nil
}

func deleteDecisionInstanceExample(ctx context.Context, client *camunda.CamundaClient) error {
	// region DeleteDecisionInstance
	return client.DeleteDecisionInstance(ctx,
		openapi.MustDecisionEvaluationKey("2251799813685310"),
		*openapi.NewDeleteDecisionInstanceRequest())
	// endregion DeleteDecisionInstance
}

func deleteDecisionInstancesBatchOperationExample(ctx context.Context, client *camunda.CamundaClient) error {
	// region DeleteDecisionInstancesBatchOperation
	req := openapi.NewDecisionInstanceDeletionBatchOperationRequest(*openapi.NewDecisionInstanceFilter())

	result, err := client.DeleteDecisionInstancesBatchOperation(ctx, *req)
	if err != nil {
		return err
	}
	fmt.Printf("created batch operation %v\n", result.GetBatchOperationKey())
	// endregion DeleteDecisionInstancesBatchOperation
	return nil
}

func searchDecisionRequirementsExample(ctx context.Context, client *camunda.CamundaClient) error {
	// region SearchDecisionRequirements
	result, err := client.SearchDecisionRequirements(ctx, *openapi.NewDecisionRequirementsSearchQuery())
	if err != nil {
		return err
	}
	for _, d := range result.GetItems() {
		fmt.Printf("%v\n", d)
	}
	// endregion SearchDecisionRequirements
	return nil
}

func getDecisionRequirementsExample(ctx context.Context, client *camunda.CamundaClient) error {
	// region GetDecisionRequirements
	drd, err := client.GetDecisionRequirements(ctx, openapi.MustDecisionRequirementsKey("2251799813685320"))
	if err != nil {
		return err
	}
	fmt.Printf("%v\n", drd)
	// endregion GetDecisionRequirements
	return nil
}

func getDecisionRequirementsXMLExample(ctx context.Context, client *camunda.CamundaClient) error {
	// region GetDecisionRequirementsXML
	xml, err := client.GetDecisionRequirementsXML(ctx, openapi.MustDecisionRequirementsKey("2251799813685320"))
	if err != nil {
		return err
	}
	fmt.Println(xml)
	// endregion GetDecisionRequirementsXML
	return nil
}

File: .sdk-repos/orchestration-cluster-api-php/examples/decision.php
<?php

/**
 * Compilable usage examples for decision-definition, decision-instance, and DRD operations.
 */

declare(strict_types=1);

namespace Camunda\Orchestration\Examples;

use Camunda\Orchestration\CamundaClient;
use Camunda\Orchestration\Semantic\DecisionDefinitionKey;
use Camunda\Orchestration\Semantic\DecisionEvaluationInstanceKey;
use Camunda\Orchestration\Semantic\DecisionEvaluationKey;
use Camunda\Orchestration\Semantic\DecisionRequirementsKey;

// region EvaluateDecision
/**
 * Evaluate decision.
 */
function evaluate_decision(CamundaClient $client, \Camunda\Orchestration\Api\Model\DecisionEvaluationById|\Camunda\Orchestration\Api\Model\DecisionEvaluationByKey $decisionEvaluationInstruction): void
{
    $client->evaluateDecision($decisionEvaluationInstruction);
}
// endregion EvaluateDecision

// region SearchDecisionDefinitions
/**
 * Search decision definitions.
 */
function search_decision_definitions(CamundaClient $client, ?\Camunda\Orchestration\Api\Model\DecisionDefinitionSearchQuery $decisionDefinitionSearchQuery = null): void
{
    $client->searchDecisionDefinitions($decisionDefinitionSearchQuery);
}
// endregion SearchDecisionDefinitions

// region GetDecisionDefinition
/**
 * Get decision definition.
 */
function get_decision_definition(CamundaClient $client, DecisionDefinitionKey $decisionDefinitionKey): void
{
    $client->getDecisionDefinition((string) $decisionDefinitionKey);
}
// endregion GetDecisionDefinition

// region GetDecisionDefinitionXML
/**
 * Get decision definition XML.
 */
function get_decision_definition_xml(CamundaClient $client, DecisionDefinitionKey $decisionDefinitionKey): void
{
    $client->getDecisionDefinitionXML((string) $decisionDefinitionKey);
}
// endregion GetDecisionDefinitionXML

// region SearchDecisionInstances
/**
 * Search decision instances.
 */
function search_decision_instances(CamundaClient $client, ?\Camunda\Orchestration\Api\Model\DecisionInstanceSearchQuery $decisionInstanceSearchQuery = null): void
{
    $client->searchDecisionInstances($decisionInstanceSearchQuery);
}
// endregion SearchDecisionInstances

// region GetDecisionInstance
/**
 * Get decision instance.
 */
function get_decision_instance(CamundaClient $client, DecisionEvaluationInstanceKey $decisionEvaluationInstanceKey): void
{
    $client->getDecisionInstance((string) $decisionEvaluationInstanceKey);
}
// endregion GetDecisionInstance

// region DeleteDecisionInstance
/**
 * Delete decision instance.
 */
function delete_decision_instance(CamundaClient $client, DecisionEvaluationKey $decisionEvaluationKey, ?\Camunda\Orchestration\Api\Model\DeleteDecisionInstanceRequest $deleteDecisionInstanceRequest = null): void
{
    $client->deleteDecisionInstance((string) $decisionEvaluationKey, $deleteDecisionInstanceRequest);
}
// endregion DeleteDecisionInstance

// region DeleteDecisionInstancesBatchOperation
/**
 * Delete decision instances (batch).
 */
function delete_decision_instances_batch_operation(CamundaClient $client, \Camunda\Orchestration\Api\Model\DecisionInstanceDeletionBatchOperationRequest $decisionInstanceDeletionBatchOperationRequest): void
{
    $client->deleteDecisionInstancesBatchOperation($decisionInstanceDeletionBatchOperationRequest);
}
// endregion DeleteDecisionInstancesBatchOperation

// region SearchDecisionRequirements
/**
 * Search decision requirements.
 */
function search_decision_requirements(CamundaClient $client, ?\Camunda\Orchestration\Api\Model\DecisionRequirementsSearchQuery $decisionRequirementsSearchQuery = null): void
{
    $client->searchDecisionRequirements($decisionRequirementsSearchQuery);
}
// endregion SearchDecisionRequirements

// region GetDecisionRequirements
/**
 * Get decision requirements.
 */
function get_decision_requirements(CamundaClient $client, DecisionRequirementsKey $decisionRequirementsKey): void
{
    $client->getDecisionRequirements((string) $decisionRequirementsKey);
}
// endregion GetDecisionRequirements

// region GetDecisionRequirementsXML
/**
 * Get decision requirements XML.
 */
function get_decision_requirements_xml(CamundaClient $client, DecisionRequirementsKey $decisionRequirementsKey): void
{
    $client->getDecisionRequirementsXML((string) $decisionRequirementsKey);
}
// endregion GetDecisionRequirementsXML
